src/main.rs

```diff
@@ -208,19 +208,86 @@ fn package_crate(config: &config::Config, repository_path: &str, new_version: &s
+    match git2::Repository::open(repository_path) {
+        Ok(repo) => repo,
+        Err(e) => {
+            logger::stderr(format!("Could not open the git repository: {:?}", e));
+            process::exit(1);
+        }
+    }
+}
 
-    let args: Args = Docopt::new(USAGE)
-        .and_then(|d| d.decode())
-        .unwrap_or_else(|e| e.exit());
+fn get_repository_path(args: &Args) -> String {
+    let path = Path::new(&args.flag_path);
+    let path = fs::canonicalize(path)
+        .unwrap_or_else(|_| print_exit!("Path does not exist or a component is
+                                                            not a directory"));
+    let repo_path = path.to_str().unwrap_or_else(|| print_exit!("Path is not valid unicode"));
+    repo_path.to_string()
+}
 
-    let mut config_builder = ConfigBuilder::new();
+fn get_signature<'a>(repository_path: String) -> git2::Signature<'a> {
+    let repo = get_repo(&repository_path);
+    let signature = match git::get_signature(&repo) {
+        Ok(sig) => sig,
+            Err(e) => {
+                logger::stderr(format!("Failed to get the committer's name and email address: {}", e.description()));
+                logger::stderr(COMMITTER_ERROR_MESSAGE);
+                process::exit(1);
+            }
+    };
 
-    if args.flag_version {
-        println!("semantic.rs 🚀 -- v{}", VERSION);
-        process::exit(0);
+    signature.to_owned()
+}
+
+fn get_user_and_repo(repository_path: &str) -> (Option<String>, Option<String>) {
+    let repo = get_repo(repository_path);
+    let remote_or_none = repo.find_remote("origin");
+    match remote_or_none {
+        Ok(remote) => {
+            let url = remote.url().expect("Remote URL is not valid UTF-8").to_owned();
+            let (user, repo_name) = user_repo_from_url(&url)
+                .unwrap_or_else(|e| print_exit!("Could not extract user and repository name from URL: {:?}", e));
+
+            (Some(user), Some(repo_name))
+        },
+        Err(err) => {
+            logger::warn(format!("Could not determine the origin remote url: {:?}", err));
+            logger::warn("semantic-rs can't push changes or create a release on GitHub");
+            (None, None)
+        }
+    }
+}
+
+fn get_github_token(repository_path: &str) -> Option<String> {
+    let repo = get_repo(repository_path);
+    let remote_or_none = repo.find_remote("origin");
+    match remote_or_none {
+        Ok(remote) => {
+            let url = remote.url().expect("Remote URL is not valid UTF-8").to_owned();
+            if github::is_github_url(&url) {
+                match env::var("GH_TOKEN") {
+                    Ok(token) => Some(token),
+                    Err(_) => None
+                }
+            } else {
+                None
+            }
+        },
+        Err(_) => None
+    }
+}
+
+fn get_cargo_token() -> Option<String> {
+    match env::var("CARGO_TOKEN") {
+        Ok(token) => Some(token),
+        Err(_) => None
```

Comment:
This whole function can become `env::var("CARGO_TOKEN").ok()` 

See [the documentation](https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.ok)